controllers/awsmachine_controller.go

```diff
@@ -323,11 +327,16 @@ func (r *AWSMachineReconciler) reconcileNormal(ctx context.Context, machineScope
 
 	switch instance.State {
 	case infrav1.InstanceStateRunning:
-		machineScope.Info("Machine instance is running", "instance-id", *machineScope.GetInstanceID())
+		machineScope.Info("EC2 instance is running", "state", instance.State, "instance-id", *machineScope.GetInstanceID())
 		machineScope.SetReady()
 	case infrav1.InstanceStatePending:
-		machineScope.Info("Machine instance is pending", "instance-id", *machineScope.GetInstanceID())
+		machineScope.Info("EC2 instance is pending", "state", instance.State, "instance-id", *machineScope.GetInstanceID())
+		r.Recorder.Eventf(machineScope.AWSMachine, corev1.EventTypeWarning, "InstancePending", "EC2 instance is %q", instance.State)
+	case infrav1.InstanceStateStopping, infrav1.InstanceStateStopped:
+		machineScope.Info("EC2 instance is hibernating", "state", instance.State, "instance-id", *machineScope.GetInstanceID())
+		r.Recorder.Eventf(machineScope.AWSMachine, corev1.EventTypeWarning, "InstanceHibernation", "EC2 instance is %q", instance.State)
 	default:
```

Comment:
Can we add a comment here that states the unexpected states and that we're going to transition to failed state?

Reply:
Done. Explicitly stating the states, and making sure the entire lifecycle is covered. Added a link to the place in the documentation describing the states as well.